lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                # exception handling below.
+                raise
+            except Exception:
+                idxctx.logger.exception("Unexpected setup error")
+                res = 12
+            else:
+                # No exceptions while processing tar ball, success.
+                res = 0
+            finally:
+                if idxctx:
+                    idxctx.dump_opctx()
+                idxctx.logger.debug("stopped processing list of tar balls")
+
+                idx = _count_lines(indexed)
+                skp = _count_lines(skipped)
+                err = _count_lines(erred)
+
+                idxctx.logger.info(
+                    "{}.{}: indexed {:d} (skipped {:d}) results," " {:d} errors",
+                    self.name,
+                    idxctx.TS,
+                    idx,
+                    skp,
+                    err,
+                )
+
+                if err > 0:
+                    if skp > 0:
+                        subj = (
+                            f"{self.name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d}"
+                            f" results, w/ {err:d} errors"
+                        )
+                    else:
+                        subj = (
+                            f"{self.name}.{idxctx.TS} - Indexed {idx:d} results, w/ {err:d}"
+                            " errors"
+                        )
+                else:
+                    if skp > 0:
+                        subj = f"{self.name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d} results"
+                    else:
+                        subj = f"{self.name}.{idxctx.TS} - Indexed {idx:d} results"
+
+                report_fname = Path(tmpdir, f"{self.name}.{idxctx.TS}.report")
+                with report_fname.open(mode="w") as fp:
+                    print(subj, file=fp)
+                    if idx > 0:
+                        print("\nIndexed Results\n===============", file=fp)
+                        with indexed.open() as ifp:
+                            for line in sorted(ifp):
+                                print(line.strip(), file=fp)
+                    if err > 0:
+                        print(
+                            "\nResults producing errors" "\n========================",
+                            file=fp,
+                        )
+                        with erred.open() as efp:
+                            for line in sorted(efp):
+                                print(line.strip(), file=fp)
+                    if skp > 0:
+                        print("\nSkipped Results\n===============", file=fp)
+                        with skipped.open() as sfp:
+                            for line in sorted(sfp):
+                                print(line.strip(), file=fp)
+                try:
+                    report.post_status(tstos(idxctx.time()), "status", report_fname)
+                except SigTermException:
```

Comment:
Do we need to handle SigTermException here?

Reply:
yes, to not get confused with general exception

Reply:
@ndokos, see the comment on lines 486 and 487. If there is a better way to word that comment to make this requirement clear, let me know.